src/rgen/Microsoft.Macios.Bindings.Analyzer/Validators/ClassValidator.cs

```diff
@@ -1,20 +1,158 @@
+	/// </summary>
+	/// <param name="properties">The properties to validate.</param>
+	/// <param name="context">The root context for validation.</param>
+	/// <param name="diagnostics">When this method returns, contains diagnostics for any duplicate strong delegate names; otherwise, an empty array.</param>
+	/// <param name="location">The code location to be used for the diagnostics.</param>
+	/// <returns><c>true</c> if all strong delegate names are unique; otherwise, <c>false</c>.</returns>
+	bool StrongDelegatesAreUnique (ImmutableArray<Property> properties, RootContext context,
+		out ImmutableArray<Diagnostic> diagnostics, Location? location = null)
+	{
+		diagnostics = ImmutableArray<Diagnostic>.Empty;
+		// use a dictionary to track all the strong names and the properties that use them
+		var strongNames = new Dictionary<string, List<Property>> ();
+		foreach (var p in properties) {
+			var strongDelegate = p.ToStrongDelegate ();
+			if (strongNames.TryGetValue (strongDelegate.Name, out var list)) {
+				list.Add (p);
+			} else {
+				// add list with the current property since we want to use is as a ref
+				strongNames.Add (strongDelegate.Name, [p]);
+			}
+		}
+		// get all the strong names that have more than one property using them
+		var duplicates = strongNames.Where (x => x.Value.Count > 1).ToImmutableArray ();
+		if (duplicates.Length == 0) {
+			// no duplicates, we are good
+			return true;
+		}
+		// build the diagnostics
+		var builder = ImmutableArray.CreateBuilder<Diagnostic> ();
+		foreach (var duplicate in duplicates) {
+			// add a diagnostic for each duplicate strong delegate using the first one as a reference and the second
+			// one as the location of the error. We use the first one as a reference because we have to choose one and
+			// is the one on top of the file
+			var firstProperty = duplicate.Value.First ();
+			for (var index = 1; index < duplicate.Value.Count; index++) {
+				var dupProperty = duplicate.Value [index]; // used for the msg and the location
+				builder.Add (Diagnostic.Create (
+					descriptor: RBI0033,
+					location: dupProperty.Location,
+					messageArgs: [
+						dupProperty.Name,
+						duplicate.Key,
+						firstProperty.Name
+					]));
+			}
+		}
+		diagnostics = builder.ToImmutable ();
+		return diagnostics.Length == 0;
+	}
+
+	/// <summary>
+	/// Validates that selectors are unique across all properties and methods in a binding.
+	/// </summary>
+	/// <param name="binding">The binding to validate.</param>
+	/// <param name="context">The root context for validation.</param>
+	/// <param name="diagnostics">When this method returns, contains diagnostics for any duplicate selectors; otherwise, an empty array.</param>
+	/// <param name="location">The code location to be used for the diagnostics.</param>
+	/// <returns><c>true</c> if all selectors are unique; otherwise, <c>false</c>.</returns>
+	bool SelectorsAreUnique (Binding binding, RootContext context,
```

Comment:
Selectors can actually be duplicated, it's only a problem if they're exported, which doesn't happen if the [Sealed] attribute is present.

Here's an example:

https://github.com/dotnet/macios/blob/052a9e91d1c7d63c946a7de51ddc76390e730cd5/src/foundation.cs#L5378-L5388

Reply:
I think we can leave it as a warning that can be disabled with a #prama so that we know it is a not common situation.